### ISSUE TITLE
KIF/KIF.h file not found

### ISSUE THREAD

**Issue by janapejic** [kif-framework/KIF#765]
I am trying to integrate KIF into my AppleTV project. I'm using Xcode 7.2, programming in Swift and integrating KIF as download from GitHub. I followed all the steps in README, including the "Use with Swift" part, importing KIF in my bridging header and setting the path to this bridging header in Build Settings of my testing target. However when I try to run the tests with ⌘U (I did not really write any yet even, using the default implementation) I get the following two errors:
- Tests-Bridging-Header.h -> 'KIF/KIF.h' file not found
- Failed to import bridging header '/path/Tests-Bridging-Header.h'

I assume that the second error will disappear once the first one is solved, but I do not know how to solve it. Any ideas?

**Comment by janapejic** [comment]
Additionally, I tried running the example project Testable Swift and this does not work either. First of all some of the syntax has to be changed, and there are still errors there:

![screen shot 2016-01-07 at 14 52 35](https://cloud.githubusercontent.com/assets/5858394/12171999/5514bd0c-b54e-11e5-91e3-908cc74e061f.png)

**Comment by kenji21** [comment]
This error can be solved adding -framework IOKit to linker flags

**Comment by janapejic** [comment]
Ah, okay, that is correct, thanks.

**Comment by phatmann** [comment]
Can you post the contents of your bridging header here please?

**Comment by janapejic** [comment]
Well, for now it is a copy of the bridging header from Testable Swift project, i.e. it only contains the line #import <KIF/KIF.h>

**Comment by phatmann** [comment]
Take a look at the header search paths in your build settings. Somehow KIF.h is not in the search path.

**Comment by miguelcampiao** [comment]
Having the exact same issue. None of the proposed workarounds is helping.

**Comment by melvynhills** [comment]
I have the same issue in an Objective-C project. I followed the Readme instructions for the Github installation.

**Comment by melvynhills** [comment]
Fixed it by adding `$(SRCROOT)/Frameworks/KIF` to Build Settings > Search Paths >  Header Search Paths, and then `#import "KIF.h"` instead of `#import <KIF/KIF.h>`

**Comment by phatmann** [comment]
@janapejic, @miguelcampiao: does the suggestion from @melvynhills fix your issue?

**Comment by miguelcampiao** [comment]
Nope. It did not fix it.

I ended up creating a project from scratch, adding KIF as a pod, and it worked.
The rest of the settings were exactly the same, to the best of my knowledge.

**Comment by melvynhills** [comment]
Additional info: make sure to choose `recursive` next to `$(SRCROOT)/Frameworks/KIF`

**Comment by afabbro** [comment]
I'm having this same issue with XCode 7.2 except I'm installing by way of CocoaPods. My project is written in Swift, and was making successful use of a Bridging Header to use another framework written in Obj-C just fine prior to trying to install KIF.

None of the proposed workarounds solve the problem.

For the other folks in this thread that have this problem, are you also on XCode 7.2? Has anyone tried to use a 7.3 beta to see if this makes a difference?

Maintainers: what other information can I give you to help diagnose the problem?

**Comment by mtitolo** [comment]
I'm having a similar issue. Question to others: Are you trying to import/use KIFUITestActor anywhere?

**Comment by afabbro** [comment]
I had a bit of a eureka moment looking at #809. I realized that I had followed the docs to the letter and that was my problem. Some of the KIF instructions are not entirely clear for usage with Swift. Namely, there is one section about installing via CocoaPods, and another for 'Using Swift', but neither section really covers the case of 'Using CocoaPods with Swift' very clearly.

I had been building KIF as a framework (`use_frameworks!` in my Podfile etc) via CocoaPods and without thinking much about it I followed the instructions where it told me to put `#import <KIF/KIF.h>` in my Bridging Header as described in the docs, under  the 'Use with Swift' section of the installation instructions:

`Your test bundle's bridging header will need to #import <KIF/KIF.h>, since KIF is a static library and not a header.`

If KIF is built as a Framework via CocoaPods then it shouldn't be included in the Bridging Header (there are some circumstances where this might be false, but largely the advice in [this article](http://www.innerexception.com/2015/05/cocoapods-useframeworks-means-bridging.html) seems reasonable).

When I removed KIF from the Bridging Header, I tried to build the test target again and I received the same error as in #809. I solved the public vs. private issue the same way as @mtitolo did while waiting on PR #810 since this is/was a huge blocker for me, and now everything seems golden.

I hope that helps folks using CocoaPods stuck on this!

**Comment by IamVicky** [comment]
Is this issue resolved ? Am still facing the issue with Xcode's new iOS UI Bundle & iOS Unit Test bundle (Objective-C). I am not using cocoa pods but installed through git submodule.
If I try the solution given by @melvynhills by adding header search paths. Then the same error is popping up in KIF library headers wherever <KIF/KIF.h>  is used. 

@mtitolo Yes I am using KIFUITestActor to create category of it and add 
- (void)navigateToLoginPage;
 Like described in https://github.com/kif-framework/KIF

My error is displayed in the image below
![screen shot 2016-03-17 at 8 18 05 pm](https://cloud.githubusercontent.com/assets/15322170/13849903/fca0dda6-ec7d-11e5-80fa-c0f9c4be2b1a.png)

**Comment by mikelupo** [comment]
Only works with Unit test targets. Not UI test target. Could that be the issue?

**Comment by IamVicky** [comment]
@mikelupo  No I tried installing KIF in Unit test targets by creating new target. But I ended up facing the same issue.

**Comment by timendez** [comment]
I'm having the same issue, ObjC by importing the framework without cocoapods

**Comment by phatmann** [comment]
@timendez: did you try any of the fixes in the comments above?

**Comment by timendez** [comment]
@phatmann Yes, they didn't work. I ended up solving my issue by simply re-adding the KIF.framework to Link Binary With Libraries in Build Phases. I could drill down command-clicking on the `#import <KIF/KIF.h>` line, so something was linked up first.

**Comment by erustique** [comment]
I need to access the camera to scan photos and so I need to run on the device. 

I have already tested on the iPhoneSimulator with this project, just fine, so I think I'm linking correctly. I added arm64 in the Info.plist file under UIRequiredDeviceCapabilities, either way, still won't build right. Ideas?

Ld build/Intermediates/MyApp.build/Debug-iphoneos/MyAppTests.build/Objects-normal/arm64/MyAppTests normal arm64
    cd ./MyApp
    export IPHONEOS_DEPLOYMENT_TARGET=10.1
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch arm64 -bundle -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.1.sdk -L./MyApp/build/Products/Debug-iphoneos -L./MyApp/build/Products/Debug-iphonesimulator -F./MyApp/build/Products/Debug-iphoneos -F./MyApp -F./MyApp/MyAppTests -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -filelist ./MyApp/build/Intermediates/MyApp.build/Debug-iphoneos/MyAppTests.build/Objects-normal/arm64/MyAppTests.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -miphoneos-version-min=10.1 -dead_strip -bundle_loader ./MyApp/build/Products/Debug-iphoneos/MyApp.app/MyApp -Xlinker -object_path_lto -Xlinker ./MyApp/build/Intermediates/MyApp.build/Debug-iphoneos/MyAppTests.build/Objects-normal/arm64/MyAppTests_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos -Xlinker -add_ast_path -Xlinker ./MyApp/build/Intermediates/MyApp.build/Debug-iphoneos/MyAppTests.build/Objects-normal/arm64/MyAppTests.swiftmodule -ObjC -framework XCTest -framework IOKit -framework XCTest -framework IOKit ./MyApp/build/Products/Debug-iphoneos/libKIF.a -Xlinker -dependency_info -Xlinker ./MyApp/build/Intermediates/MyApp.build/Debug-iphoneos/MyAppTests.build/Objects-normal/arm64/MyAppTests_dependency_info.dat -o ./MyApp/build/Intermediates/MyApp.build/Debug-iphoneos/MyAppTests.build/Objects-normal/arm64/MyAppTests

ld: warning: ignoring file ./MyApp/MyAppTests/IOKit.framework/IOKit, missing required architecture arm64 in file ./MyApp/MyAppTests/IOKit.framework/IOKit (2 slices)
Undefined symbols for architecture arm64:
  "_IOHIDEventCreateDigitizerEvent", referenced from:
      _kif_IOHIDEventWithTouches in libKIF.a(IOHIDEvent+KIF.o)
  "_IOHIDEventCreateDigitizerFingerEventWithQuality", referenced from:
      _kif_IOHIDEventWithTouches in libKIF.a(IOHIDEvent+KIF.o)
  "_IOHIDEventSetIntegerValue", referenced from:
      _kif_IOHIDEventWithTouches in libKIF.a(IOHIDEvent+KIF.o)
  "_IOHIDEventAppendEvent", referenced from:
      _kif_IOHIDEventWithTouches in libKIF.a(IOHIDEvent+KIF.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by phatmann** [comment]
Since this is a different issue, can you please file a new ticket? Thanks.

**Comment by phatmann** [comment]
Closing, since it is inactive and workarounds have been found.

**Comment by LeoTuring** [comment]
@janapejic 
how to solve this issue ?
I have the same problem, I don't konw how to solve it.

**Comment by reallyyun** [comment]
if you use cocoaPod set up KIF,  you should go to project settings -> Header Search Paths,  add "${PODS_ROOT}/KIF/Classes.   Then in your test file  #import "KIF.h"

**Comment by justinseanmartin** [comment]
With CocoaPods, you should be able to just use `#import <KIF/KIF.h>` and not need to set any special header search paths. You could also try upgrading to the latest version of CocoaPods and give that a shot.
